tests.py

```diff
@@ -89,6 +89,40 @@ def test_recursive_camelize_with_ignored_fields_and_keys(self):
         }
         self.assertEqual(camelize(data, ignore_fields=ignore_fields, ignore_keys=ignore_keys), output)
 
+    def test_preserve_underscore_keys(self):
```

Comment:
The documentation indicated that nested keys would be preserved as well, so I'd like to see 3 more examples:

data:
```
"snake_first": {"camelSecond": 1},
"snake_first": {"snake_second": 1},
"camelFirst": {"snake_second": 1},
```

output:
```
"snake_first": {"camelSecond": 1},
"snakeFirst": {"camelSecond": 1},
"snake_first": {"snake_second": 1},
"snakeFirst": {"snakeSecond": 1},
"camelFirst": {"snake_second": 1},
"camelFirst": {"snakeSecond": 1},
```

Reply:
Test updated, although the actual behavior is different than what you've outlined.  For one, you can't have the same key with different values in the dictionary.    I modified the input and expected output somewhat.  The test passes with these:

Input:

```python
        {
            "snake_first": {"camelSecond": 1},
            "snake_first_2": {"snake_second": 1},
            "camelFirst": {"snake_second": 1},
        }
```

Expected output:

```python
        {
            "snakeFirst": {"camelSecond": 1},
            "snake_first": {"camelSecond": 1},
            "snakeFirst2": {"snakeSecond": 1, "snake_second": 1},
            "snake_first_2": {"snakeSecond": 1, "snake_second": 1},
            "camelFirst": {"snakeSecond": 1, "snake_second": 1},
        }
```

I also added a setting to prevent the middleware from transforming `request.GET` from camelCase to snake_case.  I think it was doing this to make sure everything is normalized, and flows through the same transformation logic, but I didn't want that.

With all these changes, we can use it and our test suite passes.